modules/system/socket-handlers.js

```diff
@@ -1,97 +1,93 @@
 import ActorWfrp4e from "../actor/actor-wfrp4e.js";
+import WeaponDialog from "../apps/roll-dialog/weapon-dialog.js";
 import EffectWfrp4e from "./effect-wfrp4e.js";
+import WFRP_Utility from "./utility-wfrp4e.js";
 
 export default class SocketHandlers  {
 
     static call(type, payload, userId)
     {
+        if (userId == "GM") {
+            userId = game.users.activeGM.id;
+        }
         game.socket.emit("system.wfrp4e", {type, payload, userId});
     }
 
     static register()
     {
-        game.socket.on("system.wfrp4e", data => 
+        game.socket.on("system.wfrp4e", async data => 
         {
-            this[data.type]({...data.payload}, data.userId);
-        });
-    }
+            if (data.userId != game.user.id && data.userId != "ALL") return;
```

Comment:
Shouldn't the case of `userId` being empty be the same as `ALL`? There are plenty of old socket requests in the code that don't specify `userId`, wouldn't this line break all of them? Or is there something I'm missing that handles that.

Reply:
i think i can fallback the lack of userid into ALL.

Reply:
Added fallback